packages/ui/src/helpers/authenticator/__tests__/utils.test.ts

```diff
@@ -74,3 +79,24 @@ describe('configureComponent', () => {
     );
   });
 });
+
+describe('isValidEmail', () => {
+  it('should return true for a valid email address', () => {
+    expect(isValidEmail('test@example.com')).toBe(true);
+    expect(isValidEmail('TEST@EXAMPLE.COM')).toBe(true);
+  });
+
+  it('should return false for an invalid email address', () => {
```

Comment:
Nittish: Rather than running multiple tests that assert the condition within a single block the `jest` idiomatic `it.each` can be used for descriptive failures as each test runs independently, as well as improved readability of log output:
```
it.each([
  { desc: 'no domain name', email: 'test@' },
  { desc: 'an invalid domain', email: 'test@.' },
  // ...other test scenarios
])('should return false for an email address with %s', ({ email }) => {
  expect(email).toBe(false);
});

Reply:
I think the test is simple enough that it doesn't need "descriptive failures", it reads easier for me than `it.each` but ultimately don't feel super strong about it, could have a follow up to refactor other tests where `it.each` could be used. One observation I have is that currently this test runs in 1ms with all assertions, having a separate test for each assertion means each run in 1ms (I tested this). While it is negligible, I think we should pay attention to run times as well, as it can add up if this (`it.each`) is the pattern we want to adopt going forward.

Reply:
We already use `it.each` throughout the codebase. Adding 1 ms per test is fine if we are able to locate failing test case without having to comment out each test individually to find the failure (hypothetical situation). That being said this was a nit, so feel free to ignore it

Reply:
I'll add an asana task!